odoo/modules/registry.py

```diff
@@ -467,23 +470,32 @@ def check_indexes(self, cr, model_names):
                    [tuple(row[0] for row in expected)])
         existing = {row[0] for row in cr.fetchall()}
 
-        if not self.has_trigram and any(row[3] == 'trigram' for row in expected):
-            self.has_trigram = sql.install_pg_trgm(cr)
-
-        for indexname, tablename, columnname, index in expected:
+        for indexname, tablename, column_name, index, unaccent in expected:
             assert index in ('btree', 'btree_not_null', 'trigram', True, False, None)
             if index and indexname not in existing:
+                column_expression = f'"{column_name}"'
                 method = 'btree'
                 operator = ''
                 where = ''
                 if index == 'btree_not_null':
-                    where = f'"{columnname}" IS NOT NULL'
+                    where = f'{column_expression} IS NOT NULL'
                 elif index == 'trigram' and self.has_trigram:
                     method = 'gin'
                     operator = 'gin_trgm_ops'
+                    # add `unaccent` to the trigram index only because the
+                    # trigram indexes are mainly used for (i/=)like search and
+                    # unaccent is added only in these cases when searching
+                    if unaccent and self.has_unaccent:
+                        if self.has_unaccent == FunctionStatus.INDEXABLE:
+                            column_expression = get_unaccent_wrapper(cr)(column_expression)
+                        else:
+                            warnings.warn(
```

Comment:
Should probably be a logging warning.

Reply:
These days we prefer `warnings.warn` because on some platforms (Odoo.sh), using `logging.warning` just fills up the servers' disk with a lot of log files where some messages are repeated over and over.

Reply:
Sounds like entirely the wrong solution to a problem sh needs to fix anyway?

Reply:
![nojoy](https://user-images.githubusercontent.com/7571158/188116736-95e91203-3af2-4c5c-ab62-297f014068e1.jpg)